vizro-core/src/vizro/models/_components/ag_grid.py

```diff
@@ -44,10 +44,17 @@ class AgGrid(VizroBaseModel):
     # Component properties for actions and interactions
     _output_component_property: str = PrivateAttr("children")
 
-    # validator
+    # Re-used validators
     set_actions = _action_validator_factory("cellClicked")
     _validate_callable = validator("figure", allow_reuse=True, always=True)(_process_callable_data_frame)
 
+    @validator("figure")
```

Comment:
As far as I can tell, reusable validators with a parameter don't work: https://github.com/pydantic/pydantic/discussions/2938

but happy to be educated here :)

Reply:
You need don't need to pass an argument but can just use the values :) And then you can re-use the validator